apps/desktop/src/lib/components/Board.svelte

```diff
@@ -46,48 +45,49 @@
 {:else if !$branches}
 	<FullviewLoading />
 {:else}
-	<div class="board">
+	<div
+		class="board"
+		role="group"
+		on:drop={(e) => {
+			if (!dragged) return;
+			if (!sortedBranches) return;
+			dragged.style.opacity = '1';
+			e.preventDefault();
+			const updates = sortedBranches.map((b, i) => {
+				return { id: b.id, order: i };
+			});
+			branchController.updateBranchOrder(updates);
+		}}
+	>
 		<div
 			role="group"
 			class="branches"
 			data-tauri-drag-region
 			bind:this={dropZone}
 			on:dragover={(e) => {
-				if (!dragged) return;
-
 				e.preventDefault();
+				if (!dragged) throw new Error('No element to drag');
+
 				const children = [...e.currentTarget.children];
+				const currentPosition = children.indexOf(dragged);
 				dropPosition = 0;
-				// We account for the NewBranchDropZone by subtracting 2
+
+				// Calculate new index of dragged element.
 				for (let i = 0; i < children.length - 1; i++) {
+					console.log(dragged.offsetWidth / 2);
 					const pos = children[i]?.getBoundingClientRect() as DOMRect;
-					if (e.clientX > pos.left + dragged.offsetWidth / 2) {
+					if (e.clientX > pos.left + pos.width / 2) {
 						dropPosition = i + 1;
 					} else {
 						break;
 					}
 				}
-				const idx = children.indexOf(dragged);
-				if (idx !== dropPosition) {
-					if (idx >= dropPosition) {
-						children[dropPosition]?.before(dragged);
-					} else {
-						children[dropPosition]?.after(dragged);
-					}
-				}
-			}}
-			on:drop={(e) => {
-				if (!dragged) return;
-				if (!$branches) return;
-				dragged.style.opacity = '1';
-				e.preventDefault();
-				if (priorPosition !== dropPosition) {
-					const el = $branches.splice(priorPosition, 1);
-					$branches.splice(dropPosition, 0, ...el);
-					const updates = $branches.map((b, i) => {
-						return { id: b.id, order: i };
-					});
-					branchController.updateBranchOrder(updates);
+
+				// Update sorted branch array manually.
+				if (currentPosition !== dropPosition) {
+					const el = sortedBranches.splice(currentPosition, 1);
+					sortedBranches.splice(dropPosition, 0, ...el);
+					sortedBranches = sortedBranches; // Redraws #each loop.
```

Comment:
I'm a little surprised that this works, and the prior line doesn't.

In vue3, you'd have to say something like `sortedBranches = [...sortedBranches]` because it would only react to a change in object reference. Svelte however seems to behave much better with methods like `push` which is in part what surprises me about `splice` not working.

Reply:
Looking it up. I seem to be wrong, and what you did there is reasonable svelte code.

Reply:
A'ight. I but also, I put this PR back to draft. I want to fix one more thing, a once and for all, we've had too many issues with branch re-ordering.